### DISCUSSION TITLE
SolidJS adapter

### DISCUSSION THREAD

**Discussion by jcs224** [inertiajs/inertia#1310] [Ideas]
It's probably a long shot asking for first-party support for yet-another front-end framework :sweat_smile: but I thought I'd put it out into the universe. I've tried to create an adapter myself, just a little out of my league.

SolidJS is a lightweight framework with performance almost on-par with vanilla JS, and a small bundle size to boot. It's also wonderfully easy to use!

https://www.solidjs.com/

**Comment by Rastian95** [comment]
I was also looking for the same, but, without success.
I asked Ryan Carniato directly and he told me that is possible but the don't know enough about the particulars,
some agnostic solutions he would found difficult with Solid but he thinks one like that wouldn't be that different than say how they work with Astro. He suggest to ask in discord channels.
Maybe with need to ask directly too Jonathan Reinink

**Comment by superspike7** [comment]
it would be really cool to have an adapter for solidjs.

**Reply by rodsfreitas** [reply]
yeah, it would be awesome!

**Comment by jpmaga** [comment]
I had a poc around for a while, just put it in a repository. https://github.com/jpmaga/inertia-solid-adapter

Feel free to clone, fork, play with it. Have a ball. Again, it is a POC, from the little I used it, it seems to work fine, but expect bugs. No SSR support or anything. At least not for now, and I don't really use it, so probably that will need to be someone else adding it.

**Comment by nathandaly** [comment]
Bump.

**Comment by iksaku** [comment]
Hello everyone, I saw this discussion existed recently, but I wanted to comment when I finished my part.

I have also been working on a SolidJS adapter, just like @jpmaga did, although I would've loved to find his work before I started hacking another implementation 😅.

I've just published my adapter for SolidJS here: https://github.com/iksaku/inertia-adapter-solid.

If you're interested, my adapter is based on the adapters for all three other frameworks (React, Svelte, Vue), trying to incorporate the best integration ideas from each while also trying to keep as close to SolidJS practices as possible (may not be "best" ones but what I ultimately felt "correct").

There's some exciting stuff I integrated into my adapter, which I hope  you will like:
- [`@solidjs/meta`](https://github.com/solidjs/solid-meta) has been adopted as the official way for my adapter to work with SEO/Metadata because it works wonderfully in CSR and SSR, and the implementation was relatively straight-forward.
- Simple SSR configuration, just like in Svelte's adapter, not requiring to define a Server-side Renderer function. To achieve this, my adapter uses Solid's `renderToString()` API. I didn't see value in allowing you to choose other APIs like `renderToStringAsync()` or `renderToStream()`, but I would love to hear your feedback if you are interested in using these APIs.
- `useForm()` and `useRemember()` both make use of [SolidJS Store](https://docs.solidjs.com/references/api-reference/stores/using-stores) APIs, trying to keep as close as possible to Solid's DX: https://twitter.com/iksaku2/status/1653141039409160192.
- `usePage()` also uses Stores under the hood for that juicy, fine-grained reactivity.

Please let me know if you have a chance to test it out and encounter any issues or if you have any ideas that could help make this a better integration.

🫡

**Reply by jpmaga** [reply]
Hey @iksaku, I had a look at your implementation and it definitely looks more solid (pun intended) than mine. Haven't tried it yet, but will try to give it a go in the next couple of days. Also in 3 weeks, give or take, I might have a project where i can test this in a more real world scenario.

Seems pretty fleshed out, but if you need help with anything, let me know.

**Reply by iksaku** [reply]
@jpmaga thank you! Would love to hear your feedback when you try it out!

I feel like there's some stuff that could be ironed out based on people using it, so let's wait and see 😄

**Reply by iksaku** [reply]
Hello everyone! 
A few days ago I noticed that Persistent Layouts not working properly in my adapter, then went to develop a really weird workaround for it.

Before merging in, I made a post in Solid's discord looking for help reviewing the implementation hoping that somebody can give pointers for a better implementation. I would love if anyone can join the discussion there and provide your feedback on the code, or if you can get to a better approach, let me know!

https://discord.com/channels/722131463138705510/1108166822286602361/1108166822286602361

> Btw, the core of the implementation (with minimal reproduction) is available as a Playground linked in the post, as well as the link to the PR of my repo.

**Reply by iksaku** [reply]
For those interested, a few days ago I tagged [v0.2](https://github.com/iksaku/inertia-adapter-solid/releases/tag/0.2.0) of my adapter, mainly updating support for SolidJS ^1.8 and SolidMeta ^0.29.
I have refrained to tag a 1.0 as I'm currently not using the adapter in any production environments (😭) and I'm not aware of any  pitfalls in the adapter (not enough feedback I think?).
Regardless, I will continue supporting the adapter because I love SolidJS.
Feedback, bug reports and PRs are always welcome!

**Comment by mt-shihab26** [comment]
SolidJS is a kind of UI framework that makes sense to me. It is closer to web standards, more performant, and enjoyable to use. Also, I am a big fan of Laravel + Inertia's productivity. So first-party SolidJS support will be a big win for future web development. 

So, I am also requesting the Inertia.js core dev team to address this issue

**Comment by darrenbarklie** [comment]
Also arrived here, looking for official support. Great to see some in the community exploring solutions!

**Reply by joaofranciscoguarda** [reply]
Another one requesting here 🙋

**Comment by ollivr** [comment]
Solidjs Support would be amazing! I literally can't go back to react or vue. Workarounds exist, but would love to see 1st party support!

**Comment by brunoanken** [comment]
I would love SolidJS support too. With it, Inertia + Solid would certainly be my go to stack for pretty much everything when I have the privilege of choice.

**Comment by yourjhay** [comment]
Would love to see SolidJS support too..  Instead of creating an API just to use SolidJS on the front-end 😄 
for now I'm trying https://github.com/iksaku/inertia-adapter-solid for personal projects.

**Reply by iksaku** [reply]
@yourjhay happy to hear feedback from it. Been a bit busy with life so project is behind a lot, but with more feedback it would be easier for me to continue improving it

**Comment by punyflash** [comment]
I've created a SolidJS adapter in my alternative library for Inertia - https://github.com/westacks/vortex. The docs are still in progress, but core features include EVERYTHING that Inertia has. Also, my library is much more customizable than inertia.

**Comment by youfun** [comment]
look for  SolidJS support too

**Comment by vimal-360** [comment]
I keep coming across this page while looking for a solution. I want JSX support without React. Ideally, I’m looking for something like Inertia.js for SolidJS, or at least some guidance on building a custom adapter.

**Comment by erfanmola** [comment]
Up

**Comment by iksaku** [comment]
Hello everyone! Sorry for the rude ping, but I'm pretty close to tag the first major release of my own SolidJS adapter for InertiaJS!

If anyone is interested, please help me test and open issues on the repo as I want to tag the first major soon!
* Repo: https://github.com/iksaku/inertia-adapter-solid.


P.S:
> I'm currently targeting Inertia v2, so not sure if I'm going semantic Adapter v1 = Inertia v2, or if I'll be skipping Adapter v1, so Adapter v2 = Inertia v2. Thoughts on this are also welcome!

**Reply by iksaku** [reply]
After this release, I would love to start working on Inertia v3 compatibility as it comes with a bunch of shiny new stuff to play with ✨ 

But one step at a time 👶
